### PR TITLE
[Vertex AI] Add environment variable to control integration tests

### PR DESCRIPTION
**Pull request by andrewheard**
- Skip Vertex AI integration tests when the `VertexAIRunIntegrationTests` environment variable is not set.
- Switched `spm-integration` to run on iOS.
- Cleaned up dependencies in the `FirebaseVertexAIIntegration` test target.

#no-changelog

### CODE REVIEW

**Comment by paulb777** [comment]
Not sure it worked:

Executed 2 tests, with 2 tests **skipped** and 0 failures (0 unexpected) in 0.016 (0.019) seconds

https://github.com/firebase/firebase-ios-sdk/actions/runs/9420258231/job/25951894734#step:6:151

File: .github/workflows/vertexai.yml [hunk 1]
```diff
@@ -43,13 +43,14 @@ jobs:
...
       matrix:
-        target: [macOS]
+        target: [iOS]
         os: [macos-14]
         include:
           - os: macos-14
             xcode: Xcode_15.2
     runs-on: ${{ matrix.os }}
     env:
+      TEST_RUNNER_VertexAIRunIntegrationTests: 1
```
Comment (andrewheard) [inline .github/workflows/vertexai.yml:53]: TIL you need to prefix with `TEST_RUNNER_` for it to make it into the environment variables in the XCTest:

> `xcodebuild` now supports passing certain environment variables to test runner processes. In the environment where `xcodebuild` is invoked, prefix any variable with `TEST_RUNNER_` to pass that variable (with the prefix stripped) to XCTest test runner processes. For example, running `env TEST_RUNNER_Foo=Bar xcodebuild test ...` causes the environment variable `Foo=Bar` to be set in the test runner’s environment. (74104870)

-- https://developer.apple.com/documentation/xcode-release-notes/xcode-13-release-notes#Testing

Although I am using Xcode 15.2, not 15.3, I stumbled across this from https://forums.developer.apple.com/forums/thread/749185.

**Comment by andrewheard** [comment]
> Not sure it worked:
> 
> Executed 2 tests, with 2 tests **skipped** and 0 failures (0 unexpected) in 0.016 (0.019) seconds
> 
> [firebase/firebase-ios-sdk/actions/runs/9420258231/job/25951894734#step:6:151](https://github.com/firebase/firebase-ios-sdk/actions/runs/9420258231/job/25951894734#step:6:151)

Thanks for catching this, @paulb777! I think the approach worked when I was testing it in #13087 because it was running directly on macOS. Looks better now:
- ["Executed 2 tests, with 0 failures"](https://github.com/firebase/firebase-ios-sdk/actions/runs/9422799532/job/25959896874?pr=13095#step:6:155) in `spm-integration`.
- ["Executed 2 tests, with 2 tests skipped"](https://github.com/firebase/firebase-ios-sdk/actions/runs/9422799543/job/25959868533?pr=13095#step:7:3908) in `swift-build-run`.

File: .github/workflows/vertexai.yml (hunk 1 above)
Comment (paulb777) [reply .github/workflows/vertexai.yml:53]: Good find! TIL, too!

**Review by paulb777 — APPROVED** [review]

**Pull request merged**